### PR TITLE
Adding User&UserProfile to api

### PR DESCRIPTION
**Pull request by Artur-Leshko**

### CODE REVIEW

File: api/urls.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from django.urls import path, include
+from rest_framework_simplejwt.views import (
+    TokenObtainPairView,
+    TokenRefreshView,
+    TokenVerifyView
+)
+from .views import signup
+
+urlpatterns = [
```
Comment (euseand) [inline api/urls.py:9]: all patterns should follow next rule:
hostname/api/v1/route-name

File: api/urls.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    # AUTH
+    path('register/', signup, name='user_registration'),
```
Comment (euseand) [inline api/urls.py:11]: /api/v1/auth/register

File: api/urls.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    path('token/', TokenObtainPairView.as_view(), name='token_obtain_pair'),
```
Comment (euseand) [inline api/urls.py:12]: /api/v1/auth/sign-in

File: api/urls.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    path('token/refresh/', TokenRefreshView.as_view(), name='token_refresh'),
+    path('token/verify/', TokenVerifyView.as_view(), name='token_verify'),
+
+    # USER PROFILE
+    path('userprofile/', include('api.userprofile.urls')),
```
Comment (euseand) [inline api/urls.py:17]: /api/v1/user/<user_id>/profile

File: api/userprofile/serializers.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    '''
+
+    is_admin = serializers.ReadOnlyField()
+
+    class Meta:
+        '''
+            Meta class for User serializer
+        '''
+        model = User
+        fields = ['username', 'email', 'is_admin']
```
Comment (euseand) [inline api/userprofile/serializers.py:19]: should we provide user email in our api? seems unsecure

File: api/userprofile/views.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        '''
+        userprofile = self.get_object(request.user)
+        # implenet put method
+
+    def delete(self, request):
+        '''
+            Deletes UserProfile for logged in user
+        '''
+        userprofile = self.get_object(request.user)
+        userprofile.delete()
```
Comment (euseand) [inline api/userprofile/views.py:47]: simply delete profile? so we can have user accounts without profile in application?

File: api/userprofile/views.py [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+    def get_object(self, pk):
+        '''
+            Returns UserProfile for public user
+        '''
+
+        try:
+            userprofile = UserProfile.objects.get(pk=pk)
+        except UserProfile.DoesNotExist:
+            raise Http404
```
Comment (euseand) [inline api/userprofile/views.py:67]: you should provide some custom exceptions and responses

File: api/views.py [hunk 8]
```diff
@@ -1,3 +1,32 @@
...
+
+@csrf_exempt
+@transaction.atomic
+def signup(request):
+    if request.method == 'POST':
+        try:
+            data = JSONParser().parse(request)
+
+            with transaction.atomic():
+                user = User.objects.create(email=data['email'], username=data['username'])
```
Comment (euseand) [inline api/views.py:21]: what if the email or username already occupied?

File: api/views.py [hunk 9]
```diff
@@ -1,3 +1,32 @@
...
+                user.set_password(data['password'])
+                user.save()
+
+                userprofile = UserProfile.objects.create(user=user, id=user.id)
+                userprofile.save()
+
+                refreshToken = RefreshToken.for_user(user)
+
+            return JsonResponse({ 'refresh': str(refreshToken), 'access': str(refreshToken.access_token), })
```
Comment (euseand) [inline api/views.py:30]: shouldn't it happen on sign-in?

File: api/views.py [hunk 10]
```diff
@@ -1,3 +1,32 @@
...
+        except IntegrityError:
```
Comment (euseand) [inline api/views.py:31]: why wrap most of the method while you can raise custom exception?

File: userprofile/models.py [hunk 11]
```diff
@@ -1,3 +1,121 @@
...
+
+    def create_user(self, username, email, password, **extra_fields):
+        '''
+            function that handle user creating
+        '''
+
+        if not email:
+            raise ValueError("Email address is required field!")
+        elif not username:
+            raise ValueError("Useranem is required field!")
```
Comment (euseand) [inline userprofile/models.py:30]: typo

**Review by euseand — CHANGES_REQUESTED** [review]
provide solutions following the comments

File: api/urls.py (hunk 1 above)
Comment (Artur-Leshko) [reply api/urls.py:9]: added api version

File: api/urls.py (hunk 2 above)
Comment (Artur-Leshko) [reply api/urls.py:11]: changed to /api/v1/auth/sign-up/

File: api/urls.py (hunk 4 above)
Comment (Artur-Leshko) [reply api/urls.py:17]: path was changed

File: api/userprofile/serializers.py (hunk 5 above)
Comment (Artur-Leshko) [reply api/userprofile/serializers.py:19]: remove email from UserProfile serializer

File: api/userprofile/views.py (hunk 6 above)
Comment (Artur-Leshko) [reply api/userprofile/views.py:47]: now deleting User model that leads deleting UserProfile model

File: api/userprofile/views.py (hunk 7 above)
Comment (Artur-Leshko) [reply api/userprofile/views.py:67]: implemented CustomApiExeption class

File: api/views.py (hunk 10 above)
Comment (Artur-Leshko) [reply api/views.py:31]: method was rewritten

File: api/views.py (hunk 9 above)
Comment (Artur-Leshko) [reply api/views.py:30]: method was rewritten, now it returns only success message

**Review by euseand — APPROVED** [review]

**Pull request merged**